src/main/java/marquez/resources/JobResource.java

```diff
@@ -3,56 +3,112 @@
   }
 
-  @POST
+  @PUT
+  @Path("/{job}")
   @Consumes(APPLICATION_JSON)
   @Timed
-  public Response create(final Job job) {
-    dao.insert(job);
-    return Response.created(buildURI(Job.class, job.getName())).build();
+  public Response create(
+      @PathParam("namespace") final String namespace,
+      @PathParam("job") final String job,
+      @Valid CreateJobRequest request)
+      throws ResourceException {
+    try {
+      if (!namespaceExists(namespace)) {
+        return Response.status(Response.Status.NOT_FOUND).build();
+      }
+      Job jobToCreate =
+          apiJobToCoreJobMapper.map(
+              new marquez.api.Job(
+                  job,
+                  null,
+                  request.getInputDatasetUrns,
+                  request.getOutputDatasetUrns,
+                  request.getLocation(),
+                  request.getDescription()));
+      Job createdJob = jobService.createJob(namespace, jobToCreate);
+      return Response.status(Response.Status.CREATED)
+          .entity(coreJobToApiJobMapper.map(createdJob))
+          .build();
+    } catch (UnexpectedException e) {
+      LOG.error(e.getLocalizedMessage());
+      throw new ResourceException();
+    }
   }
 
   @GET
+  @Path("/{job}")
   @Timed
-  public Response getAll(@DefaultValue(DEFAULT_LIMIT) @QueryParam("limit") final int limit) {
-    return Response.ok(dao.findAll(limit)).build();
+  public Response getJob(
+      @PathParam("namespace") final String namespace, @PathParam("job") final String job)
+      throws ResourceException {
+    try {
+      Optional<Job> returnedJob = jobService.getJob(namespace, job);
+      if (returnedJob.isPresent()) {
+        return Response.status(Response.Status.OK)
+            .entity(Entity.json(coreJobToApiJobMapper.map(returnedJob.get())))
+            .build();
+      }
+      return Response.status(Response.Status.NOT_FOUND).build();
+    } catch (UnexpectedException e) {
+      LOG.error(e.getLocalizedMessage());
+      throw new ResourceException();
+    }
   }
 
   @GET
-  @Path("/{name}")
   @Timed
-  public Response getJob(@PathParam("name") final String name) {
-    return Response.ok(dao.findByName(name)).build();
+  public Response listJobs(@PathParam("namespace") final String namespace)
+      throws ResourceException {
+    // TODO: Deal with the case of an invalid namespace
+    try {
+      List<Job> jobList = jobService.getAllJobsInNamespace(namespace);
+      return Response.status(Response.Status.OK).entity(new ListJobsResponse(jobList)).build();
+
+    } catch (UnexpectedException e) {
+      LOG.error(e.getLocalizedMessage());
+      throw new ResourceException();
+    }
   }
 
-  @PUT
-  @Path("/{name}")
-  @Consumes(APPLICATION_JSON)
-  @Timed
-  public Response update(@PathParam("name") final String jobName) {
-    return Response.ok().build();
+  public boolean namespaceExists(String namespace) throws UnexpectedException {
+    return namespaceService.get(namespace).isPresent();
```

Comment:
note: Shawn will add a direct method call for this.

Reply:
if you rebase off of issue-100, the `NamespaceService` is there and you can use that `exists()` method